python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -122,62 +122,53 @@ async def get_chat_message_contents(
         Returns:
             A list of chat message contents.
         """
+        if (
+            settings.function_choice_behavior is None
+            or not settings.function_choice_behavior.auto_invoke_kernel_functions
+        ):
+            return await self._send_chat_request(chat_history, settings)
+
+        self._verify_function_choice_behavior(settings, **kwargs)
+        kernel: Kernel = kwargs.get("kernel")
+        arguments: KernelArguments = kwargs.get("arguments")
+        self._configure_function_choice_behavior(settings, kernel)
+
+        for request_index in range(settings.function_choice_behavior.maximum_auto_invoke_attempts):
+            completions = await self._send_chat_request(chat_history, settings)
+            chat_history.add_message(message=completions[0])
+            function_calls = [item for item in chat_history.messages[-1].items if isinstance(item, FunctionCallContent)]
+            if (fc_count := len(function_calls)) == 0:
+                return completions
+
+            results = await self._process_function_calls(
+                function_calls=function_calls,
+                chat_history=chat_history,
+                kernel=kernel,
+                arguments=arguments,
+                function_call_count=fc_count,
+                request_index=request_index,
+                function_behavior=settings.function_choice_behavior,
+            )
+
+            if any(result.terminate for result in results if result is not None):
+                return completions
+        else:
+            # do a final call without auto function calling
+            return await self._send_chat_request(chat_history, settings)
```

Comment:
how are the settings updated for this last one?

Reply:
Glad you commented on this. This is a good question. When we used to use `FunctionCallBehavior` there was a line of code here to set the `function_call_behavior.auto_invoke_kernel_functions = False`. This doesn't do anything to actually disable including the functions as part of the tools / tool_choice that is sent in the request body. Unless I'm completely missing it, we'd need to set `function_call_behavior.enable_kernel_functions = False` so that it skips the update to the config? Although, even with that, it would still include the old `tool_choice = 'auto'` and the previous tools that were defined.

What was the intended behavior here?

Reply:
@TaoChenOSU in the OpenAI version, there is a update to the settings to disable function calling so that you always finish up with a text response, instead of another function_call...

Reply:
Will address in another PR